flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -530,54 +515,109 @@ Actor with ActorLogMessages with ActorLogging {
         log.error(t, "Could not properly unregister job {} form the library cache.", jobID)
     }
   }
-
-  private def checkJavaVersion(): Unit = {
-    if (System.getProperty("java.version").substring(0, 3).toDouble < 1.7) {
-      log.warning("Warning: Flink is running with Java 6. " +
-        "Java 6 is not maintained any more by Oracle or the OpenJDK community. " +
-        "Flink currently supports Java 6, but may not in future releases," +
-        " due to the unavailability of bug fixes security patched.")
-    }
-  }
 }
 
 object JobManager {
+  
   import ExecutionMode._
+
   val LOG = LoggerFactory.getLogger(classOf[JobManager])
+
   val FAILURE_RETURN_CODE = 1
+
   val JOB_MANAGER_NAME = "jobmanager"
   val EVENT_COLLECTOR_NAME = "eventcollector"
   val ARCHIVE_NAME = "archive"
   val PROFILER_NAME = "profiler"
 
   def main(args: Array[String]): Unit = {
+
+    // startup checks and logging
     EnvironmentInformation.logEnvironmentInfo(LOG, "JobManager")
-    val (configuration, executionMode, listeningAddress) = parseArgs(args)
+    checkJavaVersion()
 
-      if(SecurityUtils.isSecurityEnabled) {
+    val (configuration: Configuration,
+         executionMode: ExecutionMode,
+         listeningAddress:  Option[(String, Int)]) =
+    try {
+      parseArgs(args)
+    }
+    catch {
+      case t: Throwable => {
+        LOG.error(t.getMessage(), t)
+        System.exit(FAILURE_RETURN_CODE)
+        null
```

Comment:
We have to, to silence the compiler, since it does not know that there is no return from `System.exit()` and it is not possible to assign `Nothing` to the triplet value.